### PR TITLE
Make all rules warning for RulesetKind.AllEnabled

### LINKED ISSUE #4715: Fix editorconfigs for Microsoft.CodeAnalyzis.NetAnalyzers
Comparing microsoft.codeanalysis.netanalyzers\5.0.3\editorconfig\AllRulesEnabled\.editorconfig with microsoft.codeanalysis.fxcopanalyzers\3.3.2\editorconfig\AllRulesEnabled\.editorconfig shows that, while the severity is `warning` for all diagnistics in Microsoft.CodeAnalyzis.FxCopAnalyzers, it is `silent` or `suggestion` for several diagnostics in Microsoft.CodeAnalyzis.NetAnalyzers.

That defeats **AllRulesEnabled**.

/cc @mavasani

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #4715

@mavasani I'm not sure if changes are needed for other `RulesetKind`s. For example, `CategoryEnabled`, should it enable all the category as warnings? or enable the category with the default severity?

### CODE REVIEW

**Review by Youssef1313 — COMMENTED** [review]
In the [docs](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview#enable-additional-rules), I can only see the documentation corresponding to :

- `RulesetKind.AllEnabled` (`<AnalysisMode>AllEnabledByDefault</AnalysisMode>`)
- `RulesetKind.AllDisabled` (`<AnalysisMode>AllDisabledByDefault</AnalysisMode>`)
- `RulesetKind.AllDefault` (`<AnalysisMode>Default</AnalysisMode>`).

Not sure how other options are supplied by user or where they're documented.

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 1]
```diff
@@ -130,8 +130,8 @@ public static int Main(string[] args)
 
             createRulesetAndEditorconfig(
                 "AllRulesEnabled",
-                "All Rules Enabled with default severity",
-                "All Rules are enabled with default severity. Rules with IsEnabledByDefault = false are force enabled with default severity.",
+                "All Rules Enabled as build warnings",
```
Comment (mavasani) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:133]: Need a similar change for https://github.com/dotnet/roslyn-analyzers/blob/33582d0369adb7657731b611f4811e981dcb2f7f/src/Tools/GenerateDocumentationAndConfigFiles/Program.cs#L152-L157

File: src/Tools/GenerateDocumentationAndConfigFiles/Program.cs [hunk 2]
```diff
@@ -1049,18 +1049,22 @@ string getRuleAction(DiagnosticDescriptor rule)
 
                         RulesetKind.AllDefault => getRuleActionCore(enable: rule.IsEnabledByDefault),
 
-                        RulesetKind.AllEnabled => getRuleActionCore(enable: true),
+                        RulesetKind.AllEnabled => getRuleActionCore(enable: true, enableAsWarning: true),
```
Comment (mavasani) [inline src/Tools/GenerateDocumentationAndConfigFiles/Program.cs:1052]: Need a similar change for https://github.com/dotnet/roslyn-analyzers/blob/33582d0369adb7657731b611f4811e981dcb2f7f/src/Tools/GenerateDocumentationAndConfigFiles/Program.cs#L1044

**Comment by Youssef1313** [comment]
@mavasani Thanks for review. I addressed your feedback and also made the change to CustomTagEnabled.

**Review by mavasani — APPROVED** [review]
Thanks!

**Pull request merged**